azkaban-common/src/main/java/azkaban/utils/OsCpuUtil.java

```diff
@@ -0,0 +1,179 @@
+package azkaban.utils;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import javax.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Utility class for getting CPU usage (in percentage)
+ *
+ * CPU information is obtained from /proc/stat, so only Linux systems will support this class
+ *
+ * Assumes frequent calls at regular intervals to {@link #getCpuLoad() getCpuLoad}. The length of
+ * time over which cpu load is calculated can be adjusted with parameter
+ * {@code numCpuStatsToCollect} in the class constructor and how often {@link #getCpuLoad()
+ * getCpuLoad} is called.
+ * Example: if {@link #getCpuLoad() getCpuLoad} is called every second and {@code
+ * numCpuStatsToCollect} is set to 60 then each call to {@link #getCpuLoad() getCpuLoad} returns
+ * the cpu load over the last minute.
+ */
+public class OsCpuUtil {
+
+  private static final Logger logger = LoggerFactory.getLogger(OsCpuUtil.class);
+
+  private static final String CPU_STAT_FILE = "/proc/stat";
+
+  private final Deque<CpuStats> collectedCpuStats;
+
+  @Inject
+  public OsCpuUtil(int numCpuStatsToCollect) {
+    if (numCpuStatsToCollect <= 0) {
+      numCpuStatsToCollect = 1;
+    }
+    this.collectedCpuStats = new ArrayDeque<>(numCpuStatsToCollect);
```

Comment:
Why do we need a double-ended queue here? Wil a queue be sufficient?

Reply:
We don’t really need a double-ended queue, we need a FIFO data struct which java.util.Queue doesn't guarantee according to oracle docs "Queues typically, but do not necessarily, order elements in a FIFO (first-in-first-out) manner." ArrayDeque was the most suitable FIFO implementation I could find.